src/plugins/expressions/common/expression_functions/types.ts

```diff
@@ -0,0 +1,85 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { UnwrapPromiseOrReturn } from '@kbn/utility-types';
+import { ArgumentType } from './arguments';
+import { TypeToString } from '../types/common';
+import { ExecutionContext } from '../execution/types';
+
+/**
+ * `ExpressionFunctionDefinition` is the interface plugins have to implement to
+ * register a function in `expressions` plugin.
+ */
+export interface ExpressionFunctionDefinition<
+  Name extends string,
+  Input,
+  Arguments,
+  Output,
+  Context extends ExecutionContext = ExecutionContext
```

Comment:
Same comment as below about naming- this looks wrong.

Also, `ExecutionContext` seems to be optional, right?

Reply:
> Same comment as below about naming- this looks wrong.

I've addressed that other comment&mdash;should be good now.

> Also, `ExecutionContext` seems to be optional, right?

`ExecutionContext` type in this type definition is option. But the argument in the actual function is required.